src/mlpack/methods/preprocess/CMakeLists.txt

```diff
@@ -41,3 +41,9 @@ add_markdown_docs(preprocess_imputer "cli" "preprocessing")
 add_cli_executable(preprocess_scale)
 add_python_binding(preprocess_scale)
 add_markdown_docs(preprocess_scale "cli;python" "preprocessing")
+
+if (NOT HAS_STB)
```

Comment:
Should this be `if (HAS_STB)`?  It seems like that's the situation where we'd want to compile these bindings.

Reply:
https://github.com/mlpack/mlpack/pull/2019#discussion_r352258221 

How to disabse Make this , I mean what is the option to build 

Because if I give if (HAS_STB) it does not build by default 

I am not sure whether there is an option as HAS_STB set in cmake

Reply:
Oh, sorry, I see, we are only setting `HAS_STB` as `add_definitions()`:

```
        add_definitions(-DHAS_STB)
```

(Line 330, `CMakeLists.txt`)

What you can do is, just under that line, add

```
set(HAS_STB 1)
```

and then you should be able to use `if (HAS_STB)` here. :+1:

Reply:
Here we should be able to use `STB_AVAILABLE` instead of `HAS_STB`.  I saw in the commit logs that it seemed like this didn't work, but I didn't understand what the failure was from the CI build logs.  If you want, I can try and help figure it out.